src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -759,6 +776,24 @@ private void Parse()
                 @"((?<username>[^:@]+)(:(?<password>[^:@]*))?@)?" +
                 serversPattern + @"(/" + databasePattern + ")?/?" + optionsPattern + "$";
 
+            if (_originalConnectionString.Contains("%"))
+            {
+                bool invalidPercentEncoding = false;
+#if NETSTANDARD1_5
+                var invalidPercentPattern = @"%$|%.$|%[^0-9a-fA-F]|%[0-9a-fA-F][^0-9a-fA-F]";
+                invalidPercentEncoding = Regex.IsMatch(_originalConnectionString, invalidPercentPattern);
+
+#else
+                invalidPercentEncoding = AllPercentEncodingsAreValid(_originalConnectionString);
```

Comment:
Don't you need to negate the return value of `AllPercentEncodingsAreValid`?

This should have caused a test to fail.

Reply:
That would explain why Evergreen is so unhappy :) I should have been more careful, let me fix that.